libs/wire-api-federation/src/Wire/API/Federation/BackendNotifications.hs

```diff
@@ -127,6 +190,12 @@ ensureQueue chan queue = do
 newtype FedQueueClient c a = FedQueueClient (ReaderT FedQueueEnv IO a)
   deriving (Functor, Applicative, Monad, MonadIO, MonadReader FedQueueEnv)
 
+reqOrigin :: FedQueueClient c (RequestId, Domain)
+reqOrigin = do
```

Comment:
I would just inline this.  (It's two lines instead of one, but shorter ones, so I'm not sure you've even saved *any* characters?)